src/hooks/useAxiosWithAuth0.js

```diff
@@ -23,10 +23,25 @@ export default function useAxiosWithAuth0() {
   };
 
   useEffect(() => {
-    (async () => {
-      const token = await getAccessTokenSilently();
-      axiosWithAuth(token);
-    })();
+    const locations = [
+      '/',
+      '/apply',
+      '/apply/mentee',
+      '/apply/mentor',
+      '/apply/success',
+    ];
+
+    /********************************** */
+
+    !locations.some(item => {
+      return item !== document.location.pathname;
+    }) &&
+      (async () => {
+        console.log(document.location);
+        const token = await getAccessTokenSilently();
+        axiosWithAuth(token);
+      })();
+    /********************************** */
```

Comment:
This is a really smart way to solve this problem, checking which page is being viewed and only pulling the token on pages you need to be signed in to view anyways.